src/Executables/Game/World/Map.cs

```diff
@@ -279,22 +280,46 @@ private MonsterEntity SpawnMonster(uint id, SpawnPoint spawnPoint)
             var baseY = spawnPoint.Y;
             if (spawnPoint.RangeX != 0)
             {
-                baseX += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeY);
+                baseX += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeX);
             }
 
             if (spawnPoint.RangeY != 0)
             {
-                baseY += RandomNumberGenerator.GetInt32(-spawnPoint.RangeX, spawnPoint.RangeY);
+                baseY += RandomNumberGenerator.GetInt32(-spawnPoint.RangeY, spawnPoint.RangeY);
             }
 
-            var monster = new MonsterEntity(_monsterManager, _dropProvider, _animationManager, this, _logger, _itemManager,
+            var monster = new MonsterEntity(_monsterManager, _dropProvider, _animationManager, this, _logger,
+                _itemManager,
                 id,
-                (int) (PositionX + (baseX + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) *
-                    SPAWN_POSITION_MULTIPLIER),
-                (int) (PositionY + (baseY + RandomNumberGenerator.GetInt32(-SPAWN_BASE_OFFSET, SPAWN_BASE_OFFSET)) *
-                    SPAWN_POSITION_MULTIPLIER),
-                RandomNumberGenerator.GetInt32(0, 360));
-            _world.SpawnEntity(monster);
+                0,
+                0
+            );
+
+            if (((EAiFlags)monster.Proto.AiFlag).HasAnyFlags(EAiFlags.NoMove))
+            {
+                monster.PositionX = (int)(PositionX + baseX * SPAWN_POSITION_MULTIPLIER);
+                monster.PositionY = (int)(PositionY + baseY * SPAWN_POSITION_MULTIPLIER);
+                var compassDirection = (int)spawnPoint.Direction - 1;
+                var rotation = 45 * (compassDirection < 0 ? 8 : compassDirection);
```

Comment:
magic number `8` and `45`
I would make them as `readonly static int` or something which calculates it once.

Aren't they:
* `(int)Enum.GetValues(typeof(Foo)).Cast<ESpawnPointDirection>().Last();` - I know its much more verbose but not a magic number and extensible whenever this might change?
* `360 / previousStatic`

Reply:
I realized that `8` shouldn't be there. If the direction is less than zero, it means the rotation must be random. :) I totally forgot, well spotted, thanks!